File: mobile-client/android/Contract/Order/Detail/OrderDetailPayload.kt
package app.mobiling.client.client.contract.order.detail

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
data class OrderDetailPayload(
    val orderId: String,
    val orderNumber: String,
    val stateCode: String,
    val totalLabel: String,
    val subtotalLabel: String,
    val taxationLabel: String?,
    val shippingLabel: String?,
    val placedAtIso: String,
    val lineItems: List<OrderLineItemPayload>,
)

data class OrderLineItemPayload(
    val lineId: String,
    val productId: String,
    val productTitle: String,
    val quantity: Int,
    val lineTotalLabel: String,
)

File: client/ios/Contract/Order/Detail/OrderDetailPayload.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
struct OrderDetailPayload {
    let orderId: String
    let orderNumber: String
    let stateCode: String
    let totalLabel: String
    let subtotalLabel: String
    let taxationLabel: String?
    let shippingLabel: String?
    let placedAtIso: String
    let lineItems: [OrderLineItemPayload]
}

struct OrderLineItemPayload {
    let lineId: String
    let productId: String
    let productTitle: String
    let quantity: Int
    let lineTotalLabel: String
}
